### PR TITLE
Update/extend handshake model

### PR DESCRIPTION
**Pull request by mjoyce91**
Merge https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/527 first

Adds new columns to the `BidHandshake` model
- `bidder_status` - The last status update taken by the bidder/CDO. This does not affect the more global, "source of truth" `status`. Can only be `A` or `D`
- `date_offered`
- `date_revoked`
- `date_accepted`
- `date_declined`

Another update: if a handshake is re-offered by the bureau, then the `bidder_status` is null'd out (but past dates are kept).

All dates are used for informational purposes only - they are not used for any type of business logic.

### CODE REVIEW

File: talentmap_api/bidding/views/bidhandshake.py [hunk 1]
```diff
@@ -56,14 +56,17 @@ def put(self, serializer, pk, cp_id, **ars):
 
         # Revoke any previously offered handshakes for this cp_id
         hsToArchive = BidHandshake.objects.exclude(bidder_perdet=pk).filter(cp_id=cp_id)
-        hsToArchive.update(last_editing_user=user, status='R', update_date=datetime.now())
+        hsToArchive.update(last_editing_user=user, status='R', update_date=datetime.now(), date_offered=datetime.now())
```
Comment (scott062) [inline talentmap_api/bidding/views/bidhandshake.py:59]: date_revoked?

**Comment by scott062** [comment]
For my own learning - when you re-use datetime.now() in the same .update(arg1=datetime.now(), arg2=datetime.now()), does it make sense to set that as a variable or should it calculate the same regardless? Is it any more efficient to set it before, and just pass it for both?

File: talentmap_api/bidding/views/bidhandshake.py (hunk 1 above)
Comment (mjoyce91) [reply talentmap_api/bidding/views/bidhandshake.py]: Good catch, yes

**Comment by mjoyce91** [comment]
> For my own learning - when you re-use datetime.now() in the same .update(arg1=datetime.now(), arg2=datetime.now()), does it make sense to set that as a variable or should it calculate the same regardless? Is it any more efficient to set it before, and just pass it for both?

Ah good point, because no, they will be slightly off from each other. However, it's only going to be to the [10,000th of a second](https://onlinegdb.com/dCqbdh9Jm) and I'm not sure if we gain anything from that level of granularity.

**Comment by scott062** [comment]
date_revoked isn't being set on revoke?

**Review by scott062 — APPROVED** [review]

**Pull request merged**